src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -58,21 +59,35 @@ public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
                 ExtensionParsers = _settings.ExtensionParsers
             };
 
-            // Parse the OpenAPI Document
-            var document = context.Parse(yamlDocument, diagnostic);
+            OpenApiDocument document = null;
 
-            // Resolve References if requested
-            switch (_settings.ReferenceResolution) 
+            try
+            {
+                // Parse the OpenAPI Document
+                document = context.Parse(yamlDocument, diagnostic);
+
+                // Resolve References if requested
+                switch (_settings.ReferenceResolution)
+                {
+                    case ReferenceResolutionSetting.ResolveAllReferences:
```

Comment:
maybe to use the "default: throw ..." is better.

Reply:
This chunk of code is going to change fairly significantly when adding support for resolving external files, so I'll review this in a future PR.